src/test/java/com/puppycrawl/tools/checkstyle/gui/CodeSelectorPModelTest.java

```diff
@@ -0,0 +1,111 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2016 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.gui;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.puppycrawl.tools.checkstyle.api.CheckstyleException;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.DetailNode;
+
+import com.puppycrawl.tools.checkstyle.gui.MainFrameModel.ParseMode;
+
+public class CodeSelectorPModelTest {
+
+    private MainFrameModel model;
+
+    private DetailAST tree;
+
+    private List<Integer> linesToPosition;
+
+    @Before
+    public void loadFile() throws CheckstyleException {
+        model = new MainFrameModel();
+        model.setParseMode(ParseMode.JAVA_WITH_JAVADOC_AND_COMMENTS);
+        model.openFile(new File(getPath("InputJavadocAttributesAndMethods.java")));
+        tree = ((DetailAST) model.getParseTreeTableModel().getRoot()).getFirstChild();
+        final List<Integer> modelLinesToPosition = model.getLinesToPosition();
```

Comment:
it is not obvious why such hack over lines2position is required, please put a comment , or what is better create private method and describe nuance in detail in its javadoc.